chain/chain/src/flat_storage_resharder.rs

```diff
@@ -847,20 +887,45 @@ impl FlatStorageResharder {
         self.remove_resharding_event();
     }
 
-    /// Computes the scheduling status of a task waiting to be started. The task will be ready to
-    /// start only if its resharding block has become final. Scheduling will fail if the resharding
-    /// block is ends up in a discarded fork.
+    /// Computes the scheduling status of a task waiting to be started, taking into account all possible
+    /// resharding blocks across all forks.
+    /// The task will be ready to start if one resharding block has become final.
     fn compute_scheduled_task_status(
         &self,
-        resharding_hash: &CryptoHash,
+        resharding_blocks: &[BlockInfo],
         chain_store: &ChainStore,
-    ) -> FlatStorageReshardingTaskSchedulingStatus {
+    ) -> TaskSchedulingStatus {
+        use TaskSchedulingStatus::*;
+        let mut at_least_one_postponed = false;
+        for block_info in resharding_blocks {
+            match self.compute_scheduled_task_status_at_block(&block_info, chain_store) {
+                CanStart(block_info) => return CanStart(block_info),
+                Failed => {} // Nothing to do on failure.
```

Comment:
Can you mention that this may happen during forks and isn't actually "the end of the world"?